pkg/query-service/model/queryParams.go

```diff
@@ -404,6 +404,28 @@ type TTLParams struct {
 	DelDuration           int64  // Seconds after which data will be deleted.
 }
 
+type CustomRetentionTTLParams struct {
+	Type           string                `json:"type"`
+	DefaultTTLDays int                   `json:"defaultTTLDays"`
+	ResourceRules  []CustomRetentionRule `json:"resourceRules"`
+}
+
+type CustomRetentionTTLResponse struct {
+	Message string `json:"message"`
+}
+
+type GetCustomRetentionTTLResponse struct {
+	DefaultTTLDays int                   `json:"defaultTTLDays"`
+	ResourceRules  []CustomRetentionRule `json:"resourceRules"`
+	Status         string                `json:"status"`
+}
+
+type CustomRetentionRule struct {
+	Key   string `json:"key"`
+	Name  string `json:"name"`
```

Comment:
Key = service.name
Value = application 
TTL = 90 days.

More cleaner ?

Reply:
`Value` should be the final value of the TTL, so IMO, name can be replaced to resource_key or something, but `value` can't be the `name`